server/node_modules

```diff
@@ -0,0 +1 @@
+/npm/node_modules
```

Comment:
> Running npm install locally seems like the most convenient approach, but later things like inconsistencies between Node and NPM versions sneak in. Maybe the NPM install works locally, but then fails inside the container because the version you're running locally is newer.

True, and if you build the docker container and try to run tests, it will fail.

I think it's more than a minor hack, because it changes the way that I've been developing on this to *only* work with containers, which are really a pain to debug through. I'd much rather develop against my local versions, and then run tests to make sure they work between the containers.

If the point is to allow the `run` to happen with the volume (so that it overwrites the copy I assume? Not sure why you need that), I feel like there's got to be a better way then sneakily putting things into a lump `/npm` folder on the user's root if they run `npm install` inside the project folder.